test/test.retry.ts

```diff
@@ -138,6 +138,27 @@ describe('🛸 retry & exponential backoff', () => {
     scope.done();
   });
 
+  it('should retain the baseUrl on retry', async () => {
```

Comment:
Can you check *all* the options are maintained on retry? If `baseUrl` was vulnerable, I'm wondering if the other properties are, too.

Reply:
For example, if the user passed `opts.params`, it is overwritten in the first call to `validateOpts()`: https://github.com/googleapis/gaxios/blob/d07d8320f94c275c925833ea126b802c199c8a1b/src/gaxios.ts#L156. So, then next time through, it seems like it would be extended with the same stuff.